ext/MeasurementsForwardDiffExt.jl

```diff
@@ -0,0 +1,265 @@
+module MeasurementsForwardDiffExt
+
+using ForwardDiff: Dual, DiffRules, NaNMath, LogExpFunctions, SpecialFunctions,≺
+using Measurements: Measurement, Measurements
+import Base: +,-,/,*,promote_rule
+using ForwardDiff: AMBIGUOUS_TYPES, partials, values, Partials, value
+using ForwardDiff: ForwardDiff
+
+#=
+overload in ForwardDiff.construct_seeds and ForwardDiff.single_seed
+
+A ForwardDiff.Partials is a static vector, containing the derivative component.
+construct_seeds makes a tuple of partials, each partial is full of zeros, except in the position i.
+ForwardDiff.single_seed(::Type{Partials{N},Val{i}) is the function that constructs those partials.
+
+The problem is that ForwardDiff uses one(T) to build such tuples.
+
+In Measurements.jl, one(::Type{Measurement{T}})::T has a different return type compared to zero(::Measurement{T})::Measurement{T}
+
+This causes ForwardDiff to fail, as it expects that typeof(one(T)) == typeof(zero(T)) == T.
+
+The error is in ForwardDiff, as the correct function to use is oneunit(T).

```

Comment:
To be clear, if https://github.com/JuliaDiff/ForwardDiff.jl/pull/665 was merged we wouldn't need all these methods?  It really sounds like that should be fixed upstream, not a fun of peering into internals of other packages to fix their bugs.

Reply:
Yes